packages/docs/build/bots/onchain-integrations.mdx

```diff
@@ -0,0 +1,374 @@
+  
+  await handler.sendInteractionRequest(
+    event.channelId,
+    {
+      case: 'transaction',
+      value: {
+        id: 'usdc-transfer',
+        title: 'Send USDC',
+        subtitle: 'Send 50 USDC',
+        content: {
+          case: 'evm',
+          value: {
+            chainId: '8453',
+            to: usdcAddress,
+            value: '0',
+            data: data,
+            signerWallet: undefined // User chooses wallet
+          }
+        }
+      }
+    },
+    hexToBytes(event.userId as `0x${string}`) // Optional: target specific user
+  )
+})
+```
+
+### Signature Requests
+
+Request cryptographic signatures for authentication or permissions:
+
+```ts
+import { InteractionRequestPayload_Signature_SignatureType } from '@towns-protocol/proto'
+
+bot.onSlashCommand('sign', async (handler, event) => {
+  const typedData = {
+    domain: {
+      name: 'My Towns Bot',
+      version: '1',
+      chainId: 8453,
+      verifyingContract: '0x0000000000000000000000000000000000000000'
+    },
+    types: {
+      Message: [
+        { name: 'from', type: 'address' },
+        { name: 'content', type: 'string' },
+        { name: 'timestamp', type: 'uint256' }
+      ]
+    },
+    primaryType: 'Message',
+    message: {
+      from: event.userId,
+      content: 'I agree to the terms',
+      timestamp: Math.floor(Date.now() / 1000)
+    }
+  }
+  
+  await handler.sendInteractionRequest(
+    event.channelId,
+    {
+      case: 'signature',
+      value: {
+        id: 'agreement-signature',
+        title: 'Sign Agreement',
+        subtitle: `Sign: "${typedData.message.content}"`,
+        chainId: '8453',
+        data: JSON.stringify(typedData),
+        type: InteractionRequestPayload_Signature_SignatureType.TYPED_DATA,
+        signerWallet: undefined
+      }
+    },
+    hexToBytes(event.userId as `0x${string}`)
+  )
+})
+```
+
+### Handling Responses
+
+Handle transaction and signature responses:
+
+```ts
+bot.onInteractionResponse(async (handler, event) => {
+  const { response } = event
+  
+  switch (response.payload.content?.case) {
+    case 'transaction':
+      const txData = response.payload.content.value
+      if (txData.txHash) {
+        await handler.sendMessage(
+          event.channelId,
+          `Transaction confirmed: ${txData.txHash}`
+        )
+      } else if (txData.error) {
+        await handler.sendMessage(
+          event.channelId,
+          `Transaction failed: ${txData.error}`
+        )
+      }
+      break
+      
+    case 'signature':
+      const signatureData = response.payload.content.value
+      if (signatureData.signature) {
+        await handler.sendMessage(
+          event.channelId,
+          `Signature received: ${signatureData.signature}`
+        )
+      } else if (signatureData.error) {
+        await handler.sendMessage(
+          event.channelId,
+          `Signature failed: ${signatureData.error}`
+        )
+      }
+      break
+  }
+})
```

Comment:
a better example would be tracking our bot interactions by checking if the response matches the original request id. Could be inside a switch statement